create_bot_knowledge.py

```diff
@@ -80,139 +80,107 @@ def connect_to_source(project: Tuple[str, str]):
     return service, repo
 
 
-def extract_knowledge_from_repository(project: Tuple[str, str], update_knowledge: bool = False):
+def check_directory(knowledge_dir):
+    if not knowledge_dir.exists():
+        _LOGGER.info("No knowledge from any repo has ever been created, creating new directory at %s" % knowledge_dir)
+        os.mkdir(knowledge_dir)
+
+def is_old_knowledge(project_knowledge):
+    if project_knowledge.exists():
+        _LOGGER.info("Previous collected knowledge of repo %s/%s found" % (project[1], project[0]))
+        return True
+    
+    _LOGGER.info("No previous knowledge from repo %s/%s found" % (project[1], project[0]),
+                 "New knowledge file will be created")
+
+
+def pull_analysis(pull, results):
+    commits = pull.commits
+    # TODO: Use commits to extract information.
+    # commits = [commit for commit in pull.get_commits()]
+
+    reviews = pull.get_reviews()
+    label_names = [label.name for label in pull.get_labels()]
+    author = str(pull.user.login)
+    pr_created = pull.created_at.timestamp()
+
+    # Get the review approvation if it exists
+    approvation = next((review for review in reviews if review.state == 'APPROVED'), None)
+    pr_approved = approvation.submitted_at.timestamp() if approvation is not None else None
+    pr_approved_by = str(approvation.user.login) if approvation is not None else None
+    pr_ttr = pr_approved - pr_created if approvation is not None else None
+    
+    results[str(pull.number)] = {
+        "PR_labels": label_names,
+        "PR_created": pr_created,
+        "PR_approved": pr_approved,
+        "PR_approved_by": pr_approved_by,
+        "PR_TTR": pr_ttr,
+        "PR_author": author,
+        "PR_commits_number": commits
+    }
+
```

Comment:
This function should `return results`

Reply:
should it? This way i'm just modifying the results within the function, so pull request will be inserted into the results each time i call the funciton. Or is this approach bad practice?

Reply:
you are right, I forgot to delete this comment atually, forget it :)